crypto/http/http_client.c

```diff
@@ -577,18 +586,29 @@ int OSSL_HTTP_REQ_CTX_nbio(OSSL_HTTP_REQ_CTX *rctx)
             }
         }
         if (n <= 0) {
+            if (rctx->state == OHS_ERROR_CONTENT) {
+                if (OSSL_TRACE_ENABLED(HTTP))
+                    OSSL_TRACE(HTTP, "]\n"); /* end of error response content */
+                /* in addition, throw error on inconsistent length: */
+                (void)check_set_resp_len("error response content", rctx, resp_len);
+                return 0;
+            }
             if (BIO_should_retry(rctx->rbio))
                 return -1;
             ERR_raise(ERR_LIB_HTTP, HTTP_R_FAILED_READING_DATA);
             return 0;
         }
 
         /* Write data to memory BIO */
-        if (BIO_write(rctx->mem, rctx->buf, n) != n)
+        if (BIO_write(rctx->mem, buf, n) != n)
```

Comment:
Don't get this change on first look

Reply:
This is just a simplification, also to make `BIO_*` calls more consistent.
Note that `buf` has been assigned before to `(char *)rctx->buf`.

Reply:
OK, agreed